src/app/client/src/app/modules/workspace/components/content-creation-default-template/content-creation-default-template.component.ts

```diff
@@ -0,0 +1,304 @@
+*/
+  private urlString;
+  /**
+* contentType is creation type, fected from url
+*/
+  private contentType;
+  /**
+ * userForm name creation
+ */
+  public creationForm: FormGroup;
+  /**
+* To call resource service which helps to use language constant
+*/
+  public resourceService: ResourceService;
+  /**
+* To call resource service which helps to use language constant
+*/
+  public userService: UserService;
+  /**
+* userProfile is of type userprofile interface
+*/
+  public userProfile: IUserProfile;
+  /**
+* Contains config service reference
+*/
+  public config: ConfigService;
+/**
+ * frameworkService is get framework data
+ */
+  public frameworkService: FrameworkService;
+/**
+ * formService is to get form meta data
+ */
+  public formService: FormService;
+/**
+ * formInputData is to take input data's from form
+ */
+  public formInputData: any;
+/**
+ * categoryList is category list of dropdown values
+ */
+  public categoryList: Object;
+/**
+ * masterList is master copy of framework data
+ */
+  public masterList: any;
+/**
+ * years is used to get years from getYearsForCreateTextBook
+ */
+  public years: any;
+    /**
+   * To make content editor service API calls
+   */
+  private editorService: EditorService;
+
+  // private categoryMasterList: any;
+
+
+
+  constructor(
+    formService: FormService,
+    private _cacheService: CacheService,
+    private router: Router,
+    resourceService: ResourceService,
+    frameworkService: FrameworkService,
+    toasterService: ToasterService,
+    userService: UserService,
+    config: ConfigService,
+    editorService: EditorService
+  ) {
+    this.formService = formService;
+    this.resourceService = resourceService;
+    this.frameworkService = frameworkService;
+    this.toasterService = toasterService;
+    this.categoryList = {};
+    this.userService = userService;
+    this.config = config;
+    this.editorService = editorService;
+  }
+
+  ngAfterViewInit() {
+    const DROPDOWN_INPUT_TYPES = ['select', 'multiSelect'];
+    _.forEach(this.formFieldProperties, (field) => {
+      if (_.includes(DROPDOWN_INPUT_TYPES, field.inputType)) {
+        if (field.depends && field.depends.length) {
+          this.getAssociations(this.categoryMasterList[field.code],
+            field.range, (associations) => {
+              this.applyDependencyRules(field, associations, false);
+            });
+        }
+      }
+    });
+  }
+  ngOnInit() {
+    this.formInputData = {};
+    this.masterList = {};
+    /***
+ * Call User service to get user data
+ */
+    this.userService.userData$.subscribe(
+      (user: IUserData) => {
+        if (user && !user.err) {
+          this.userProfile = user.userProfile;
+        }
+      });
+    this.showLoader = false;
+    this.years = this.getYearsForCreateTextBook();
+    this.mapMasterCategoryList(this.formFieldProperties, '');
+  }
+  /**
+ * requestBody is returned of type object
+ */
+  generateData(data) {
```

Comment:
i think generateData not used in this component,  why have you kept this function if you are not using